Kernel: add packages for common USB DVB adapters (v5)

M95D opened the pull request:
Adds DVB drivers intended to be used with tvheadend package and easily available Astrometa DVB-C/T/T2 and Hauppauge soloHD/dualHD USB DVB tuners.
This allows users to record and stream TV programs from DVB networks to tvheadend clients such as Kodi. It allows watching home TV from abroad via VPN.
The new multimedia category makes it easy to add new tuners in the future.

Changes:
- Add packages for these kernel modules:
  - DVB core (installs dvb-core.ko)
  - DVB USB v2 (installs dvb-usb-v2.ko)
  - Drivers for chips used in Astrometa DVB:
    - Rafael Micro R820T (tuner)
    - Realtek RTL28xxU, RTL2830, RTL2832
    - Panasonic MN88472 (no firmware)
    - Panasonic MN88473 (no firmware)
    - Sony CXD2841ER
  - Drivers for chips used in Hauppauge soloHD/dualHD:
    - Silicon Labs Si2157
    - Silicon Labs Si2168 (no firmware)
    - Empia EM28xx (installs em28xx.ko, em28xx-dvb.ko and tveeprom.ko)
- Added new kernel config lines for v6.6

motolav commented:
You shouldn't open new PRs for changes

xback commented:
This will probably collide with my changes which I would push tomorrow.

https://git.openwrt.org/?p=openwrt/staging/xback.git;a=summary

M95D commented:
Since I opened this pull request, more and more changes were needed to keep up with the main branch. I can't test it every single time. Please let me know if you decide to merge it.

zdz2019 commented on package/kernel/linux/modules/multimedia.mk:
```diff
@@ -0,0 +1,857 @@
+  DEPENDS:=+kmod-i2c-core +kmod-usb-core kmod-multimedia
+  FILES:= \
+	$(LINUX_DIR)/drivers/media/dvb-core/dvb-core.ko \
+	$(LINUX_DIR)/drivers/media/usb/dvb-usb-v2/dvb_usb_v2.ko
```
Need to add:
$(LINUX_DIR)/drivers/media/rc/rc-core.ko

M95D replied:
Did you encounter any error?

I know that Astrometa has an IR receiver, but remotes should be a separate package. Most users won't need a remote for their router.

zdz2019 replied:
> Did you encounter any error?
> 
> I know that Astrometa has an IR receiver, but remotes should be a separate package. Most users won't need a remote for their router.

x86_64 compilation error, need rc-core.ko to compile successfully.

M95D replied:
I can't reproduce. I need your .config to try it myself.

AFAIK, there's no dependency in kernel menuconfig between dvb and rc.
Also, your proposed change, `$(LINUX_DIR)/drivers/media/rc/rc-core.ko`, only installs the kernel module after it was successfully built. It doesn't make any sense.

zdz2019 commented:
The x86 test failed.

Collected errors:
 * pkg_hash_check_unresolved: cannot find dependency kmod-multimedia for kmod-usb-audio
 * pkg_hash_fetch_best_installation_candidate: Packages for kmod-usb-audio found, but incompatible with the architectures configured
 * opkg_install_cmd: Cannot install package kmod-usb-audio.

M95D commented:
It seems that there is an error with some imx targets.
`[...] v4l2-mem2mem.ko is missing.`
Can anyone help with a suggestion why this module isn't built and how to fix it? Thanks.

jellekaufmann commented:
I have been using these patches on an ath79 archer c7 v2 for a couple of weeks now and it works fine in combination with minisatip :)

Would be nice to see it upstreamed

Commit:
kernel-packages: move video cameras into a separate menu

Changes:
- Create a new menu: Multimedia
- Move video cameras from Video to Multimedia

Effects:
- Video menu will contain video output (GPU, framebuffer)
  and video processing (because it's done by GPU).
- Multimedia menu will contain video capture devices (cameras).

Signed-off-by: Marius Dinu <m95d+git@psihoexpert.ro>

Commit:
kernel-packages: remove redundant variables from paths

These directory names are unlikely to change. There is no need for variables. V4L2_DIR was only used once.
Also fixed some indents (spaces before tab).

Signed-off-by: Marius Dinu <m95d+git@psihoexpert.ro>

Commit:
kernel-packages: add DVB support and drivers (v5)

Adds DVB drivers intended to be used with tvheadend package and easily available Astrometa DVB-C/T/T2 and Hauppauge soloHD/dualHD USB DVB tuners.
This allows users to record and stream TV programs from DVB networks to tvheadend clients such as Kodi. It allows watching home TV from abroad via VPN.
The new multimedia category makes it easy to add new tuners in the future.

Changes:
- Add packages for these kernel modules:
  - DVB core (installs dvb-core.ko)
  - DVB USB v2 (installs dvb-usb-v2.ko)
  - Drivers for chips used in Astrometa DVB:
    - Rafael Micro R820T (tuner)
    - Realtek RTL28xxU, RTL2830, RTL2832
    - Panasonic MN88472 (no firmware)
    - Panasonic MN88473 (no firmware)
    - Sony CXD2841ER
  - Drivers for chips used in Hauppauge soloHD/dualHD:
    - Silicon Labs Si2157
    - Silicon Labs Si2168 (no firmware)
    - Empia EM28xx (installs em28xx.ko, em28xx-dvb.ko and tveeprom.ko)
- Added new kernel config lines for v6.6

Signed-off-by: Marius Dinu <m95d+git@psihoexpert.ro>

M95D commented:
I split it into 3 patches to make it to review.

M95D commented:
Fixed a recursive dependency. Ready to merge.

M95D commented:
@neheb: May I ask for a review, please, if you have the time. This PR is more than a year old. Or simply reject it, so I can close and forget about it. Thank you!

Shine- commented:
This would be a great addition to OpenWrt. I've been using Armbian for TVheadend so far, exactly because I couldn't get these Astrometa tuners to work in OpenWrt (or rather - I gave up early, since Armbian was working out-of-the-box anyway).  I suppose it wouldn't be hard to add more tuners once this is merged, based on your work (I have some Silicon Labs based ones in use as well)?

jellekaufmann commented:
Agreed. This is a very useful addition.

It still works fine with my two dvb-c tuners.

M95D commented:
Rebased.
@neheb, thanks for the review. How about a merge?

neheb commented:
If I had merge access, sure

M95D commented:
@hauke, please review & merge? Thank you!

M95D commented:
Changes since v4:
- Removed mc.ko - it wasn't strictly needed. Without mc.ko, there's no USB audio dependency anymore.
- Added support for Hauppauge soloHD and dualHD tuners. PCTV Systems tripleStick T2 may work too, as they use the same chips.

jneuhauser commented:
Thank you @M95D for not giving up this PR!
